src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java

```diff
@@ -207,16 +207,39 @@ public static void assertContainerMatchesJurisdictionAndPoBox(
         }
     }
 
-    public static void assertClassificationNewApplicationIfPaymentsArePresent(InputEnvelope envelope) {
-        if (envelope.payments != null && !envelope.payments.isEmpty()
-            && envelope.classification != Classification.NEW_APPLICATION) {
+    public static void assertPaymentsEnabledAndAllowedForClassification(
+        InputEnvelope envelope,
+        boolean paymentsEnabled,
+        List<ContainerMappings.Mapping> mappings
+    ) {
+        if (envelope.payments != null
+            && envelope.payments.size() > 0
+            && (!paymentsEnabled
+            || !isPaymentsEnabledForContainer(mappings, envelope)
+            || envelope.classification != Classification.NEW_APPLICATION
+            )) {
             throw new InvalidJourneyClassificationException(
```

Comment:
This looks incorrect exception ?

Reply:
Added a new Exception for payments `PaymentsDisabledException`